### PR TITLE
[5.3] Provide a notification channel contract

### PR DESCRIPTION
**Pull request by lucasmichot**
Provide a notification channel contract

### CODE REVIEW

File: src/Illuminate/Contracts/Notifications/Channel.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+interface Channel
+{
+    /**
+     * Send the given notification.
+     *
+     * @param  \Illuminate\Support\Collection  $notifiables
+     * @param  \Illuminate\Notifications\Notification  $notification
+     * @return void
+     */
+    public function send($notifiables, Notification $notification);
```
Comment (GrahamCampbell) [inline src/Illuminate/Contracts/Notifications/Channel.php]: Why isn't the collection typehinted?

Comment (GrahamCampbell) [inline src/Illuminate/Contracts/Notifications/Channel.php]: If we just wanted `iterable`. then update the doc to that.

Comment (lucasmichot) [inline src/Illuminate/Contracts/Notifications/Channel.php]: It was not because, it was not either in the channels classes

Comment (lucasmichot) [inline src/Illuminate/Contracts/Notifications/Channel.php]: Nope let's stick with `\Illuminate\Support\Collection`, as `\Illuminate\Notifications\ChannelManager::sendNow` will always pass a collection

File: src/Illuminate/Notifications/Channels/BroadcastChannel.php [hunk 2]
```diff
@@ -5,6 +5,7 @@
 use Illuminate\Notifications\Notification;
 use Illuminate\Contracts\Events\Dispatcher;
 use Illuminate\Notifications\Events\DatabaseNotificationCreated;
+use Illuminate\Support\Collection;
```
Comment (GrahamCampbell) [inline src/Illuminate/Notifications/Channels/BroadcastChannel.php]: Please sort by length.

**Comment by taylorotwell** [comment]
I purposefully left this off because I don't see a big need for defining one. If your class contains the `send` method it will be compatible if it doesn't it will throw an error. The interface only adds additional file loading.

**Pull request closed without merging**